features.go

```diff
@@ -65,3 +67,42 @@ func LookupFeature(word string) Feature {
 
 	return FtUnknown
 }
+
+// FeatureSet is a set of features. Go figure.
+type FeatureSet map[Feature]struct{}
+
+// FeatureSetFromMsg returns a populated FeatureSet from a RsFeatures
+func FeatureSetFromMsg(msg *Message) (fs FeatureSet, err error) {
+	if msg.Word() != RsFeatures {
+		err = errors.New("Message is not a FEATURES message")
+		return
+	}
+	fs = make(FeatureSet)
+	for _, featurestr := range msg.AsSlice()[1:] {
+		f := LookupFeature(featurestr)
+		if f == FtUnknown {
+			err = errors.New("Unknown feature: " + featurestr)
```

Comment:
Mehhh, no difference. Errorf is for formatting a string, string concatenation is hardly formatting. Will change it if you really care.